From 0ef887f97b17b28ddded973af4622dc8bfa2eb71 Mon Sep 17 00:00:00 2001
From: Naveen Kumar P <naveenkumar.p@ittiam.com>
Date: Tue, 23 Jan 2018 18:03:49 +0530
Subject: [PATCH] Fix output buffer size check

Bug: 72165027
Test: ran poc before/after

For output buffer size check, the parameter wd is set to larger
of disp_wd and disp_strd.

Change-Id: I1fc745753762b8a8e943165d0bf6525c500fb020
(cherry picked from commit ce8a8db32e9b2054c5dc119fbbec542bf8e848b6)
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63743
Signed-off-by: robinz1x <robinx.zhang@intel.com>
---
 decoder/ihevcd_utils.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/decoder/ihevcd_utils.c b/decoder/ihevcd_utils.c
index 5e27885..fef3f74 100755
--- a/decoder/ihevcd_utils.c
+++ b/decoder/ihevcd_utils.c
@@ -736,6 +736,9 @@ IHEVCD_ERROR_T ihevcd_check_out_buf_size(codec_t *ps_codec)
         return (IHEVCD_ERROR_T)IHEVCD_SUCCESS;
     }
 
+    if(ps_codec->i4_disp_strd > (WORD32)wd)
+        wd = ps_codec->i4_disp_strd;
+
     if(ps_codec->e_chroma_fmt == IV_YUV_420P)
         u4_min_num_out_bufs = MIN_OUT_BUFS_420;
     else if(ps_codec->e_chroma_fmt == IV_YUV_422ILE)
-- 
1.9.1

